File: solution/kotlin/day05/src/main/kotlin/people/Person.kt
package people

enum class PetType {
    CAT,
    DOG,
    HAMSTER,
    TURTLE,
    BIRD,
    SNAKE
}

data class Pet(val type: PetType, val name: String, val age: Int)
data class Person(val firstName: String, val lastName: String, val pets: List<Pet> = emptyList()) {
    fun addPet(petType: PetType, name: String, age: Int): Person =
        Person(firstName, lastName, pets + Pet(petType, name, age))
}

File: solution/c#/Day05/Day05/Person.cs
namespace Day05
{
    public record Person(string FirstName, string LastName, params Pet[] Pets);

    public record Pet(PetType Type, string Name, int Age);

    public enum PetType
    {
        Cat,
        Dog,
        Hamster,
        Turtle,
        Bird,
        Snake
    }
}